apikey-service/src/main/java/eu/europeana/apikey/controller/ApikeyController.java

```diff
@@ -646,9 +661,11 @@ private Apikey copyUpdateValues(Apikey apikey, ApikeyDetails apikeyUpdate){
     private void mandatoryMissing(ApikeyAction apikeyUpdate) throws ApikeyException {
         String retval = "Required parameter(s): ";
         ArrayList<String> missingList = new ArrayList<>();
-        if (null == apikeyUpdate.getFirstName()) missingList.add("'firstName'");
-        if (null == apikeyUpdate.getLastName()) missingList.add("'lastName'");
-        if (null == apikeyUpdate.getEmail()) missingList.add("'email'");
+        if (apikeyUpdate.getFirstName()==null || apikeyUpdate.getFirstName().isEmpty()) missingList.add("'firstName'");
```

Comment:
This is fine, but you could also use if (!StringUtils.isEmpty(apiKeyUpdate.getFirstName()) to achieve the same, or StringUtils.isBlank... to also check if the string contains at least 1 non white-space character